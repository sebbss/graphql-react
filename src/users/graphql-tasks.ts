import {gql} from '@apollo/client/core'


export const GET_ALL_PEOPLE = gql`
query fetchPeople($page: Int, $name: String!){
    fetchPeople(page: $page, name: $name) {
        count
        results{
            name
            height
            gender
            mass
            homeworld
        }
        
    }
}

`;

export const GET_ONE_PERSON = gql`
query person($id: String){
    person(id: $id) {
        name
        height
        gender
        mass
        homeworld
    }
}

`;

