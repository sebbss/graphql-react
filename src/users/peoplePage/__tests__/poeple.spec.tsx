import { ApolloError } from '@apollo/client'
import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import { MockedProvider } from '@apollo/client/testing';
import TestRenderer from "react-test-renderer"
import PeoplePage, { ContentDiv } from '../component'
import { GET_ALL_PEOPLE } from '../../graphql-tasks';
import { Notification } from 'baseui/notification';

describe("PeoplePage should render correctly", ()=>{
    it("should render without crushing", ()=>{
        const component = TestRenderer.create(
            <Router>
                <MockedProvider mocks={[]} addTypename={false}>
                    <PeoplePage 
                    data = {{fetchPeople: {results: [{name: "test", gender: "male", height:"20", mass: "90", homeworld: "http://google.com"}], count: 1}}}
                    loading={false}
                    error={new ApolloError({ networkError: new Error("error message") })}
                    page={1}
                    setName={()=>{}}
                    name={"test"}
                    setPage={()=>{}}
                    />

                 </MockedProvider>
            </Router>
            
        )
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
        expect(tree).toBeDefined()
    })

    it("should display error notification", async ()=>{
        const mock = {
            request: {
              query: GET_ALL_PEOPLE,
              variables: { name: 'test' },
            },
            error: new Error('An error occurred'),
          };
        const component = TestRenderer.create(
            <Router>
                <MockedProvider mocks={[mock]} addTypename={false}>
                    <PeoplePage 
                    data = {undefined}
                    loading={false}
                    error={new ApolloError({ networkError: new Error("error message") })}
                    page={1}
                    setName={()=>{}}
                    name={"test"}
                    setPage={()=>{}}
                    />

                 </MockedProvider>
            </Router>
            )
        
        const [notification] =await component.root.findAllByType(Notification);
        expect(notification).toBeDefined();
        expect(notification.props.children()).toBe("error message")
    })

    it("should render people content", async()=>{
        const mock = {
            request: {
              query: GET_ALL_PEOPLE,
              variables: {  },
            },
            result:{
                data: {fetchPeople: {results: [{name: "test", gender: "male", height:"10", mass: "90", homeworld: "http://google.com"}], count: 1}}
            }
          }; 
        
        const component = TestRenderer.create(
            <Router>
                <MockedProvider mocks={[mock]} addTypename={false}>
                    <PeoplePage 
                    data = {undefined}
                    loading={false}
                    error={new ApolloError({ networkError: new Error("error message") })}
                    page={1}
                    setName={()=>{}}
                    name={"test"}
                    setPage={()=>{}}
                    />

                 </MockedProvider>
            </Router>
        )
        const [card] = await component.root.findAllByType(ContentDiv)
        expect(card.children.length).toBe(1)
    })
})