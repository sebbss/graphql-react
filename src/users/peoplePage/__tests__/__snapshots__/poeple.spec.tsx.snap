// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PeoplePage should render correctly should render without crushing 1`] = `
<div
  className=""
>
  <div
    className=""
  >
    <div
      className=""
      data-baseweb="input"
    >
      <div
        className=""
      >
        <svg
          className=""
          data-baseweb="icon"
          title="Search"
          viewBox="0 0 24 24"
        >
          <title>
            Search
          </title>
          <path
            clipRule="evenodd"
            d="M11 6C8.79086 6 7 7.79086 7 10C7 12.2091 8.79086 14 11 14C13.2091 14 15 12.2091 15 10C15 7.79086 13.2091 6 11 6ZM5 10C5 6.68629 7.68629 4 11 4C14.3137 4 17 6.68629 17 10C17 11.2958 16.5892 12.4957 15.8907 13.4765L19.7071 17.2929C20.0976 17.6834 20.0976 18.3166 19.7071 18.7071C19.3166 19.0976 18.6834 19.0976 18.2929 18.7071L14.4765 14.8907C13.4957 15.5892 12.2958 16 11 16C7.68629 16 5 13.3137 5 10Z"
            fillRule="evenodd"
          />
        </svg>
      </div>
      <div
        className=""
        data-baseweb="base-input"
      >
        <input
          aria-activedescendant={null}
          aria-autocomplete={null}
          aria-controls={null}
          aria-describedby={null}
          aria-errormessage={null}
          aria-haspopup={null}
          aria-invalid={false}
          aria-label={null}
          aria-labelledby={null}
          aria-required={false}
          autoComplete="on"
          className=""
          disabled={false}
          inputMode="text"
          name=""
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          onKeyUp={[Function]}
          pattern={null}
          placeholder="Search For A Character"
          required={false}
          role={null}
          rows={null}
          type="text"
          value="test"
        />
        <div
          className=""
        >
          <svg
            aria-label="Clear value"
            className=""
            data-baseweb="icon"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
            title="Clear value"
            viewBox="0 0 24 24"
          >
            <title>
              Clear value
            </title>
            <path
              clipRule="evenodd"
              d="M12 20C16.4183 20 20 16.4183 20 12C20 7.58173 16.4183 4 12 4C7.58173 4 4 7.58173 4 12C4 16.4183 7.58173 20 12 20ZM10.0303 8.96967C9.73743 8.67679 9.26257 8.67679 8.96967 8.96967C8.67676 9.26257 8.67676 9.73743 8.96967 10.0303L10.9393 12L8.96967 13.9697C8.67676 14.2626 8.67676 14.7374 8.96967 15.0303C9.26257 15.3232 9.73743 15.3232 10.0303 15.0303L12 13.0607L13.9697 15.0303C14.2626 15.3232 14.7374 15.3232 15.0303 15.0303C15.3232 14.7374 15.3232 14.2626 15.0303 13.9697L13.0607 12L15.0303 10.0303C15.3232 9.73743 15.3232 9.26257 15.0303 8.96967C14.7374 8.67679 14.2626 8.67679 13.9697 8.96967L12 10.9393L10.0303 8.96967Z"
              fillRule="evenodd"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
  <div
    className=""
    data-baseweb="notification"
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    role="alert"
  >
    <div
      className=""
    >
      error message
    </div>
    <svg
      className=""
      data-baseweb="icon"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
      title="Close"
      viewBox="0 0 24 24"
    >
      <title>
        Close
      </title>
      <path
        clipRule="evenodd"
        d="M7.29289 7.29289C7.68342 6.90237 8.31658 6.90237 8.70711 7.29289L12 10.5858L15.2929 7.29289C15.6834 6.90237 16.3166 6.90237 16.7071 7.29289C17.0976 7.68342 17.0976 8.31658 16.7071 8.70711L13.4142 12L16.7071 15.2929C17.0976 15.6834 17.0976 16.3166 16.7071 16.7071C16.3166 17.0976 15.6834 17.0976 15.2929 16.7071L12 13.4142L8.70711 16.7071C8.31658 17.0976 7.68342 17.0976 7.29289 16.7071C6.90237 16.3166 6.90237 15.6834 7.29289 15.2929L10.5858 12L7.29289 8.70711C6.90237 8.31658 6.90237 7.68342 7.29289 7.29289Z"
        fillRule="evenodd"
      />
    </svg>
  </div>
  <div
    className=""
  >
    <div
      className=""
    >
      <div>
        <section
          className=""
          data-baseweb="card"
        >
          <div
            className=""
          >
            <h1
              className=""
            >
              test
            </h1>
            <div
              className=""
            >
              <div
                className=""
              >
                <a
                  href="/1"
                  onClick={[Function]}
                >
                  <a
                    className=""
                    data-baseweb="link"
                    onBlur={[Function]}
                    onFocus={[Function]}
                  >
                    Details
                  </a>
                </a>
              </div>
            </div>
          </div>
        </section>
      </div>
    </div>
  </div>
  <div
    className=""
  >
    <div
      className=""
      data-baseweb="pagination"
    >
      <button
        aria-label="previous page. current page 1 of 1"
        className=""
        data-baseweb="button"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
      >
        <div
          className=""
        >
          <svg
            className=""
            data-baseweb="icon"
            title=""
            viewBox="0 0 24 24"
          >
            <path
              clipRule="evenodd"
              d="M9 12C9 12.2652 9.10536 12.5196 9.29289 12.7071L13.2929 16.7071C13.6834 17.0976 14.3166 17.0976 14.7071 16.7071C15.0976 16.3166 15.0976 15.6834 14.7071 15.2929L11.4142 12L14.7071 8.70711C15.0976 8.31658 15.0976 7.68342 14.7071 7.29289C14.3166 6.90237 13.6834 6.90237 13.2929 7.29289L9.29289 11.2929C9.10536 11.4804 9 11.7348 9 12Z"
              fillRule="evenodd"
            />
          </svg>
        </div>
        Prev
      </button>
      <div
        className=""
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <div
          aria-controls={null}
          aria-expanded="false"
          aria-haspopup="true"
          className=""
          data-baseweb="select"
          onBlur={[Function]}
          onClick={[Function]}
        >
          <div
            className=""
            onClick={[Function]}
            onKeyDown={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
          >
            <div
              className=""
            >
              <div
                aria-selected="true"
                className=""
                disabled={false}
                value={1}
              >
                1
              </div>
              <div
                aria-activedescendant={null}
                aria-describedby={null}
                aria-disabled={false}
                aria-errormessage={null}
                aria-expanded={false}
                aria-label="Selected 1. page 1 of 1"
                aria-labelledby={null}
                aria-owns={null}
                aria-required={null}
                className=""
                onFocus={[Function]}
                role="listbox"
                tabIndex={0}
              >
                <input
                  aria-hidden={true}
                  id={null}
                  style={
                    Object {
                      "border": "none",
                      "opacity": 0,
                      "overflow": "hidden",
                      "padding": 0,
                      "width": 0,
                    }
                  }
                  tabIndex={-1}
                />
              </div>
            </div>
            <div
              className=""
            >
              <svg
                className=""
                data-baseweb="icon"
                title="open"
                viewBox="0 0 24 24"
              >
                <title>
                  open
                </title>
                <path
                  d="M12.7071 15.2929L17.1464 10.8536C17.4614 10.5386 17.2383 10 16.7929 10L7.20711 10C6.76165 10 6.53857 10.5386 6.85355 10.8536L11.2929 15.2929C11.6834 15.6834 12.3166 15.6834 12.7071 15.2929Z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <span
        aria-hidden={true}
        className=""
      >
        of 1
      </span>
      <button
        aria-label="next page. current page 1 of 1"
        className=""
        data-baseweb="button"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
      >
        Next
        <div
          className=""
        >
          <svg
            className=""
            data-baseweb="icon"
            title=""
            viewBox="0 0 24 24"
          >
            <path
              clipRule="evenodd"
              d="M9.29289 7.29289C8.90237 7.68342 8.90237 8.31658 9.29289 8.70711L12.5858 12L9.29289 15.2929C8.90237 15.6834 8.90237 16.3166 9.29289 16.7071C9.68342 17.0976 10.3166 17.0976 10.7071 16.7071L14.7071 12.7071C14.8946 12.5196 15 12.2652 15 12C15 11.7348 14.8946 11.4804 14.7071 11.2929L10.7071 7.29289C10.3166 6.90237 9.68342 6.90237 9.29289 7.29289Z"
              fillRule="evenodd"
            />
          </svg>
        </div>
      </button>
    </div>
  </div>
</div>
`;
